### ISSUE TITLE
Feature Request: Party Mode "Hand over Mic"

### ISSUE THREAD

**Issue by Michri** [Vocaluxe/Vocaluxe#221]
Party-mode "hand over microphone" (as in Playstation Singstar) would be nice to have. Building teams and then handing over the mic from one team member to the other while playing one title would be nice to have. Joker options like in challenge mode would be great to be in there as well (amount of jokers, reset amount of jokers after each turn).

**Comment by flokuep** [comment]
Yeah. I really like to add this, but currently really busy with some private stuff. There are some existing changes in my branch: https://github.com/flokuep/Vocaluxe/tree/game-modes
In this one, I started implementing game-modes. Pass-the-mic should be also possible with this.
see #181

**Comment by Michri** [comment]
I am not a real developer expert. How can I get in your game-modes?

**Comment by flokuep** [comment]
You just need to check out this branch. Follow our short HowTo https://github.com/Vocaluxe/Vocaluxe/wiki/HowTo:-Build-Vocaluxe-(Windows) and use 'my' url instad of Vocaluxe/develop-one.

**Comment by Michri** [comment]
I get an error when trying to clone a repository:

![image](https://cloud.githubusercontent.com/assets/20970069/19087623/0009bc72-8a73-11e6-914c-37fddb09e443.png)

How can it be fixed? I wonder about, that I wasn't asked for a GitHub login.

**Comment by flokuep** [comment]
It's the wrong link. If you like to login via ssh: git@github.com:flokuep/Vocaluxe.git or via https: https://github.com/flokuep/Vocaluxe.git - You can find these links in a little menu when pressing the green 'Clone or download' button.
After cloning you need to checkout game-modes branch.

**Comment by Michri** [comment]
Done!
What do I need to do next to get it run? I have now a lot of files in the source\Repos\Vocaluxe-folder of VS, but no .exe .

**Comment by Michri** [comment]
I think I found some doku on  that in the folder. Right now VS needs to addd tools.

**Comment by flokuep** [comment]
First you need to compile (There should be a Run-button with a green arrow or something like this...) - After this, Vocaluxe should start, and there should be an Vocaluxe.exe in output-folder. Make sure to select in the dropdown menu next to the run-button the right configuration...

**Comment by Michri** [comment]
![image](https://cloud.githubusercontent.com/assets/20970069/19091497/2e5e551e-8a83-11e6-956f-cf3cd79d8319.png)

Now I get 13 errors:

1>------ Build started: Project: Vocaluxe, Configuration: ReleaseWin x64 ------
1>  git : Die Benennung "git" wurde nicht als Name eines Cmdlet, einer Funktion, 
1>  einer Skriptdatei oder eines ausführbaren Programms erkannt. Überprüfen Sie 
1>  die Schreibweise des Namens, oder ob der Pfad korrekt ist (sofern enthalten), 
1>  und wiederholen Sie den Vorgang.
1>  In Zeile:1 Zeichen:78
1>  + ... _TAG) {$version = $Env:GIT_VERSION_TAG;} else {$version = git describ ...
1>  +                                                               ~~~
1>      + CategoryInfo          : ObjectNotFound: (git:String) [], CommandNotFound 
1>     Exception
1>      + FullyQualifiedErrorId : CommandNotFoundException
1>   
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Lib\Database\CHighscoreDB.cs(183,32,183,40): error CS1061: 'SPlayer' does not contain a definition for 'GameMode' and no extension method 'GameMode' accepting a first argument of type 'SPlayer' could be found (are you missing a using directive or an assembly reference?)
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Lib\Database\CHighscoreDB.cs(185,36,185,54): error CS0117: 'EGameMode' does not contain a definition for 'TR_GAMEMODE_MEDLEY'
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Lib\Database\CHighscoreDB.cs(188,36,188,52): error CS0117: 'EGameMode' does not contain a definition for 'TR_GAMEMODE_DUET'
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Lib\Database\CHighscoreDB.cs(191,36,191,57): error CS0117: 'EGameMode' does not contain a definition for 'TR_GAMEMODE_SHORTSONG'
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Lib\Database\CHighscoreDB.cs(292,40,292,58): error CS0117: 'EGameMode' does not contain a definition for 'TR_GAMEMODE_MEDLEY'
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Lib\Database\CHighscoreDB.cs(295,40,295,56): error CS0117: 'EGameMode' does not contain a definition for 'TR_GAMEMODE_DUET'
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Lib\Database\CHighscoreDB.cs(298,40,298,61): error CS0117: 'EGameMode' does not contain a definition for 'TR_GAMEMODE_SHORTSONG'
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Screens\CScreenHighscore.cs(241,44,241,55): error CS0117: 'CGame' does not contain a definition for 'GetGameMode'
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Screens\CScreenHighscore.cs(259,27,259,38): error CS0117: 'CGame' does not contain a definition for 'GetGameMode'
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Screens\CScreenHighscore.cs(265,32,265,50): error CS0117: 'EGameMode' does not contain a definition for 'TR_GAMEMODE_MEDLEY'
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Screens\CScreenHighscore.cs(269,32,269,48): error CS0117: 'EGameMode' does not contain a definition for 'TR_GAMEMODE_DUET'
1>C:\Users\14all\Source\Repos\Vocaluxe\Vocaluxe\Screens\CScreenHighscore.cs(274,32,274,53): error CS0117: 'EGameMode' does not contain a definition for 'TR_GAMEMODE_SHORTSONG'
========== Build: 0 succeeded, 1 failed, 6 up-to-date, 0 skipped ==========

**Comment by flokuep** [comment]
Have you installes Git correctly? Check if path to git.exe is set in environment variables. We need this for compiling vocaluxe.

**Comment by Michri** [comment]
I was not aware of installing Git. I now installed GitHub and was restarting VS, but still the errors occur.
In Windows10 path isn't set in environment variables. Should the path be added to the path-attribute?

**Comment by Michri** [comment]
I now added the path to the path-attribute, but still these errors occur.

**Comment by lukeIam** [comment]
@Michri 
> fatal: No names found, cannot describe anything.

It seems that you are missing tags - just open a command prompt at the Vocaluxe folder and execute:
```batch
git fetch --tags
```


For all your other errors:
I get the same errors when I try to build the game-modes branch.
@flokuep Could it be that some changes where not committed/pushed?

**Comment by flokuep** [comment]
Yes, you're totally right, forget to commit merge-conflict changes. Should work now.

**Comment by Michri** [comment]
Jetzt habe ich es irgendwie hinbekommen, dass Vocaluxe aus VS heraus startet, aber ich sehe trotzdem nur di e normalen Party Modes.

**Comment by flokuep** [comment]
Es gibt auch noch keinen klassichen Party-Modus. Du kannst aber mal in die Song-Auswahl gehen und dort in die Lied-Optionen schauen. Dort solltest du dann verschiedene Modi starten können.

**Comment by Michri** [comment]
Stimmt, super.
Allerdings gibt's dort (noch) nicht den "Reich das Mikro weiter" Modus, richtig? Wird das noch kommen?
Werdne die PlayModes noch in die PArty Sektion verschoben?

Basiert diese Version auf dem letzten Nightly Build, so dass ich diese Variante verwenden kann?

Vielen Dank schon mal.

**Comment by flokuep** [comment]
Irgendwann wenn ich Zeit finde ja. Und jap, müsste auf dem aktuellen dev-Branch basieren.

**Comment by Michri** [comment]
Noch mal was anderes: wo könnte man denn noch mal fertige Song-Steuer-/Text-Dateien finden?

**Comment by Michri** [comment]
wie kann ich denn informiert werden, wenn es eine neue Version Deines Branches gib?

**Comment by flokuep** [comment]
Die mir bekannte Song-Text Seite ist zurzeit nicht erreichbar, bezüglich Updates kannst du mir hier auf Github einfach folgen....

**Comment by flokuep** [comment]
I've continued working on game-modes-branch... Just added a (very rough) start of a classical party-mode. There are some features missing, like setting gamemodes correctly and let them decide about winner of current round. But feel free to have a look into code or just test.

**Comment by Michri** [comment]
Anything to test? How?

**Comment by lukeIam** [comment]
> Anything to test? How?

Locks like this one:
https://github.com/flokuep/Vocaluxe/commits/game-modes
But you have to compile it yourself.

**Comment by Michri** [comment]
I always gett error:

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0006	Metadata file 'C:\Users\14all\Source\Vocaluxe-game-modes\Output\Vocaluxe.exe' could not be found	VocaluxeTests	C:\Users\14all\Source\Vocaluxe-game-modes\VocaluxeTests\CSC	1	Active

**Comment by flokuep** [comment]
Are you sure that you're compiling the correct project/starting it correclty? Ensure that you have selected 'Release Win' (more details: https://github.com/Vocaluxe/Vocaluxe/wiki/HowTo:-Build-Vocaluxe-(Windows)

**Comment by Michri** [comment]
Done.
But I get errors:
![image](https://cloud.githubusercontent.com/assets/20970069/21615659/c1de459a-d1de-11e6-8b83-421e4d60ed56.png)

**Comment by lukeIam** [comment]
@Michri Happens sometimes (should be fixed with #254) https://github.com/Vocaluxe/Vocaluxe/pull/239#issuecomment-268257205

For now you can try to revert all changes to the `\Vocaluxe\Properties\AssemblyInfo.cs`  files.

If this does not solve the problem you can try to delete this line https://github.com/Vocaluxe/Vocaluxe/blob/develop/Vocaluxe/Vocaluxe.csproj#L487 and reset  AssemblyInfo.cs again.

**Comment by flokuep** [comment]
Or you ensure that your git installation is set up correctly (check if you could run git from cmd).

**Comment by lukeIam** [comment]
@flokuep is this also part of your partymode restructuring project or do we keep that separate?
Maybe we should create a meta issue for all the party mode stuff and link issues like this to them to keep the overview.

**Comment by flokuep** [comment]
Yeah, this is one of the game modes I want to add... I've added this to game mode project.
